pkg/tfbridge/provider.go

```diff
@@ -416,6 +417,25 @@ func (p *Provider) Configure(ctx context.Context,
 	return &pulumirpc.ConfigureResponse{}, nil
 }
 
+var requiredFieldRegex = regexp.MustCompile("\"(.*?)\": required field is not set")
+
+func (p *Provider) formatFailureReason(res Resource, reason string) string {
+	// If a required field is missing and the value can be set via config,
+	// extend the error with a hint to set the proper config value
+	name := requiredFieldRegex.FindStringSubmatch(reason)
+	if len(name) == 2 {
+		field := res.Schema.Fields[name[1]]
+		if field != nil && field.Default != nil {
+			if configKey := field.Default.Config; configKey != "" {
+				hint := "%s. Either set it explicitly or configure it with 'pulumi config set %s:%s <value>'."
+				return fmt.Sprintf(hint, reason, p.module, configKey)
```

Comment:
Is it the case that in general this error message reports back the TF name instead of the Pulumi name?

It's slightly orthogonal to this issue - but this may be an opportunity to fix the naming in the error message to use the Pulumi name (camelCased).  Happy if we want to do that separately - but we've already matched out the name in the regexp here which is the foundation of how we would do that.

Reply:
Yes, great point! My current code won't work for multi-word names, so that's related enough for sure. I'm changing the code to translate the property name and report it back as Pulumi name.